### ISSUE TITLE
Changing mech type, tonnage or heat sink type causes irrecoverable error when assigning critical locations

### ISSUE THREAD

**Issue by somebody1212** [MegaMek/megameklab#168]
### Environment

MegaMekLab 0.44.0
Windows 10
Java 8 (megameklablog.txt doesn't say which version, but I updated it today so it's the latest)

### Description
Describe the issue and provide screenshots ingame if able.
Provide the rulebook, page number, and text if you feel like this is a rules issue.
And provide the steps you have taken to reproduce the issue.
(Delete the above lines)

After changing tonnage, mech type, or heat sink type, the cockpit selection regularly bugs out when assigning critical locations, as shown in the attached image. In addition, this prevents any other components from being assigned to critical locations. The invalid cockpit selection and inability to assign critical locations then prevent the mech from ever being able to pass validation.

![image](https://user-images.githubusercontent.com/4204343/42424871-14835ecc-830c-11e8-8d17-0da4b8c9fcf4.png)

The following error is found in megameklablog.txt 

java.lang.IllegalArgumentException: setSelectedIndex: 0 out of bounds
	at javax.swing.JComboBox.setSelectedIndex(Unknown Source)
	at megameklab.com.ui.view.MekChassisView.refreshCockpit(MekChassisView.java:533)
	at megameklab.com.ui.view.MekChassisView.refresh(MekChassisView.java:392)
	at megameklab.com.ui.view.MekChassisView.setFromEntity(MekChassisView.java:299)
	at megameklab.com.ui.Mek.tabs.StructureTab.refresh(StructureTab.java:168)
	at megameklab.com.ui.Mek.MainUI.refreshAll(MainUI.java:214)
	at megameklab.com.util.Mech.CriticalTransferHandler.changeMountStatus(CriticalTransferHandler.java:571)
	at megameklab.com.util.Mech.CriticalTransferHandler.changeMountStatus(CriticalTransferHandler.java:553)
	at megameklab.com.util.Mech.CriticalTransferHandler.addEquipmentMech(CriticalTransferHandler.java:321)
	at megameklab.com.util.Mech.CriticalTransferHandler.importData(CriticalTransferHandler.java:461)
	at javax.swing.TransferHandler$DropHandler.drop(Unknown Source)
	at java.awt.dnd.DropTarget.drop(Unknown Source)
	at javax.swing.TransferHandler$SwingDropTarget.drop(Unknown Source)
	at sun.awt.dnd.SunDropTargetContextPeer.processDropMessage(Unknown Source)
	at sun.awt.dnd.SunDropTargetContextPeer$EventDispatcher.dispatchDropEvent(Unknown Source)
	at sun.awt.dnd.SunDropTargetContextPeer$EventDispatcher.dispatchEvent(Unknown Source)
	at sun.awt.dnd.SunDropTargetEvent.dispatch(Unknown Source)
	at java.awt.Component.dispatchEventImpl(Unknown Source)
	at java.awt.Container.dispatchEventImpl(Unknown Source)
	at java.awt.Component.dispatchEvent(Unknown Source)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Unknown Source)
	at java.awt.LightweightDispatcher.processDropTargetEvent(Unknown Source)
	at java.awt.LightweightDispatcher.dispatchEvent(Unknown Source)
	at java.awt.Container.dispatchEventImpl(Unknown Source)
	at java.awt.Window.dispatchEventImpl(Unknown Source)
	at java.awt.Component.dispatchEvent(Unknown Source)
	at java.awt.EventQueue.dispatchEventImpl(Unknown Source)
	at java.awt.EventQueue.access$500(Unknown Source)
	at java.awt.EventQueue$3.run(Unknown Source)
	at java.awt.EventQueue$3.run(Unknown Source)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.awt.EventQueue$4.run(Unknown Source)
	at java.awt.EventQueue$4.run(Unknown Source)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.awt.EventQueue.dispatchEvent(Unknown Source)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
	at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
	at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.awt.EventDispatchThread.pumpEvents(Unknown Source)

### Files

[megameklablog.txt](https://github.com/MegaMek/megameklab/files/2174202/megameklablog.txt)

**Comment by neoancient** [comment]
Could not reproduce. Please provide exact steps taken.

**Comment by somebody1212** [comment]
Step 1: Start MegaMekLab
![image](https://user-images.githubusercontent.com/4204343/42540469-723739fe-8496-11e8-9f53-2a326833813a.png)

Step 2: Change Motive Type to Quad
![image](https://user-images.githubusercontent.com/4204343/42540475-776596e6-8496-11e8-9c5b-3552b8bc35ed.png)

Step 3: Change mech tonnage
![image](https://user-images.githubusercontent.com/4204343/42540482-7caf61cc-8496-11e8-8b07-ab73f013c5f2.png)

Step 4: Attempt to assign a heat sink to any location
![image](https://user-images.githubusercontent.com/4204343/42540510-95a1639c-8496-11e8-90b8-8e4ccda14e05.png)

Step 5: Confirm that the bug has occured
![image](https://user-images.githubusercontent.com/4204343/42540521-a3eff97c-8496-11e8-84cd-40a90394a99c.png)

**Comment by somebody1212** [comment]
Just tested it again, the bug only appears to occur when entering the tonnage manually into the box. It does not occur when adjusting the tonnage with the buttons.

**Comment by neoancient** [comment]
I've been able to find one other person that can reproduce this, and not consistently, which has made it somewhat difficult to figure out an approach to fixing it. I have some ideas but can't test themself so made a special build when some extra debugging output to help me track it down. What I've attached here is just the JAR file. To try it out, replace the JAR file in a MML install (renaming it from MegaMekLab.zip to MegaMekLab.jar) and follow the steps to reproduce the bug. I need two things from the test run:
1. Does it happen with this version?
2. I need to see the log file whether it happens or not.
[MegaMekLab.zip](https://github.com/MegaMek/megameklab/files/2220448/MegaMekLab.zip)
